osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySpinnerApproachCircle.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Skinning.Legacy
+{
+    public class LegacySpinnerApproachCircle : CompositeDrawable
+    {
+        private DrawableSpinner drawableSpinner;
+
+        [CanBeNull]
+        private Sprite approachCircle;
+
+        [BackgroundDependencyLoader]
+        private void load(DrawableHitObject drawableHitObject, ISkinSource source)
+        {
+            drawableSpinner = (DrawableSpinner)drawableHitObject;
+
+            AutoSizeAxes = Axes.Both;
+
+            var spinnerProvider = source.FindProvider(s =>
+                s.GetTexture("spinner-circle") != null ||
+                s.GetTexture("spinner-top") != null);
+
+            if (spinnerProvider is DefaultLegacySkin)
+                return;
+
+            InternalChild = approachCircle = new Sprite
```

Comment:
At which point this whole class jsut becomes a sprite and I'm not even sure it needs to exist. Will leave that up to you to discern.

Reply:
It still has the scaling transform, I think we still want that there as that would simplify moving `ApproachCircle` to `LegacyNew` and `LegacyOld` without having to duplicate logic there (if we replace this with a `Sprite` instead).

Reply:
They still share a parent class. That's not too relevant to the discussion; it depends how the changes affect usage in mods etc. and which feels cleaner.

Reply:
Right, since the provider lookup logic has been shortened to only be present in one class and check against a single texture, I think it makes sense to remove this and do the transform in the parent class.